feature/orchestrator/src/main/java/com/simprints/feature/orchestrator/OrchestratorViewModel.kt

```diff
@@ -209,29 +211,37 @@ internal class OrchestratorViewModel @Inject constructor(
         result: Serializable,
     ) {
         if (currentStep.id == StepId.FACE_CAPTURE && result is FaceCaptureResult) {
-            val matchingStep = steps.firstOrNull { it.id == StepId.FACE_MATCHER }
+            val captureParams = currentStep.params?.let { it as? FaceCaptureParams }
+            val matchingStep = steps.firstOrNull { step ->
+                if (step.id != StepId.FACE_MATCHER) {
+                    false
+                } else {
+                    val stepSdk = step.params?.let { it as? MatchStepStubPayload }?.faceSDK
+                    stepSdk == captureParams?.faceSDK
```

Comment:
Interesting - had we completely missed this part previously?

Reply:
I was as confused when I noticed that. It probably never came out since we did not have 2 Face SDKs to be enabled at the same time before. Even now, this could ever be an issue in the dev build if you enable ROC and SimFace.